components/LanguageSwitcher.jsx

```diff
@@ -10,6 +10,15 @@ const LanguageSwitchLink = ({ locale, ...rest }) => {
   const theme = useTheme()
   const { t } = useTranslation('common')
 
+  const handleLanguageChange = (newLocale) => {
+    localStorage.setItem('appLocale', newLocale);
+    languageDetector.cache(newLocale);
+
+    // Trigger a custom event
```

Comment:
Either remove the comment, or describe precisely why a CustomEvent is being triggered.

Reply:
Fixed in https://github.com/TUM-Dev/TUMenu/pull/110/commits/972b7d8ca37be9e33adeda3e0367e6b305a155f4